feature(embed): adds elgg/embed AMD module

hypeJunction opened the pull request:
Adds elgg/embed module that is defined as dependency of a menu item.
Deprecates elgg.embed JS library

hypeJunction committed:
feature(embed): adds elgg/embed AMD module

Adds elgg/embed module that is defined as dependency of a menu item.
Deprecates elgg.embed JS library

hypeJunction commented:
Partially replaces #9684

hypeJunction commented on docs/guides/javascript.rst:
```diff
@@ -711,6 +711,9 @@ Available hooks
 **ajax_response_data, \***
     This filters the response data returned to users of the ``elgg/Ajax`` module. See :doc:`ajax` for details.
 
+**insert, editor**
```
This just documents an existing hook

hypeJunction commented:
Rebased

mrclay commented:
Do we not need code that makes sure the lightbox CSS gets loaded? Should we put that logic into elgg/lightbox so just requiring the module is sufficient?

hypeJunction commented:
I think I wrote this when my lightbox PR was loading css by default. I think we changed that later. Let's just load that bloody lightbox css on all pages and be done with it.

mrclay commented:
👍 We already load lightbox CSS on every page in 2.x

mrclay commented on mod/embed/start.php:
```diff
@@ -49,52 +55,30 @@ function embed_longtext_menu($hook, $type, $items, $vars) {
-			'href' => elgg_normalize_url($url),
-		]),
-		'text' => $text,
+		'href' => elgg_normalize_url($url),
```
This link leads to a dead end page that doesn't let you do anything, hence the previous markup. We don't want the user landing on that page if the event gets bound too late or there's a JS error.

mrclay replied:
HTML is lacking a good element for this use case. We can't use `button` because it would submit the form. `<a href="javascript:">` is sufficient.

hypeJunction replied:
I might have sneaked in data-href binding into lightbox amd
On May 17, 2016 8:31 PM, "Steve Clay" <notifications@github.com> wrote:

> In mod/embed/start.php
> <https://github.com/Elgg/Elgg/pull/9747#discussion_r63578158>:
>
> >  	}
> >
> >  	$items[] = ElggMenuItem::factory(array(
> >  		'name' => 'embed',
> > -		'href' => 'javascript:void()',
> > -		'data-colorbox-opts' => json_encode([
> > -			'href' => elgg_normalize_url($url),
> > -		]),
> > -		'text' => $text,
> > +		'href' => elgg_normalize_url($url),
>
> HTML is lacking a good element for this use case. We can't use button
> because it would submit the form. <a href="javascript:"> is sufficient.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/Elgg/Elgg/pull/9747/files/1f1dad12c0ce30ef426893802f46dd1c5f5114e1#r63578158>
>

mrclay commented:
With https://github.com/hypeJunction/Elgg/pull/8 I'm happy to pull this in, squashed or not.

mrclay committed:
chore(embed): remove href from embed button (#8)

The "embed" endpoint is not a standalone page, and the user should never load
it or be sent there if JS fails.

hypeJunction commented:
Merged but no time to squash
